src/exampleTypes/Date.js

```diff
@@ -3,22 +3,206 @@ import { observer } from 'mobx-react'
 import { Flex } from '../layout/Flex'
 import { exampleTypes } from 'contexture-client'
 import injectTreeNode from '../utils/injectTreeNode'
+import F from 'futil-js'
+import _ from 'lodash/fp'
+import moment from 'moment'
+
+let allRollingOpts = [
+  { type: 'all', label: 'All Dates', value: { from: '', to: '' } },
+  { type: 'past', label: 'Last 3 Days', value: { from: 'now-3d', to: 'now' } },
+  { type: 'past', label: 'Last 7 Days', value: { from: 'now-7d', to: 'now' } },
+  {
+    type: 'past',
+    label: 'Last 30 Days',
+    value: { from: 'now-30d', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 90 Days',
+    value: { from: 'now-90d', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 180 Days',
+    value: { from: 'now-180d', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 12 Months',
+    value: { from: 'now/d-12M', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 24 Months',
+    value: { from: 'now/d-24M', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 36 Months',
+    value: { from: 'now/d-36M', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 48 Months',
+    value: { from: 'now/d-48M', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last 60 Months',
+    value: { from: 'now/d-60M', to: 'now' },
+  },
+  {
+    type: 'past',
+    label: 'Last Calendar Month',
+    value: { from: 'now-1M/M', to: 'now/M-1d' },
+  },
+  {
+    type: 'past',
+    label: 'Last Calendar Quarter',
+    value: { from: 'lastQuarter', to: 'lastQuarter' },
+  },
+  {
+    type: 'past',
+    label: 'Last Calendar Year',
+    value: { from: 'now-1y/y', to: 'now/y-1d' },
+  },
+  {
+    type: 'present',
+    label: 'This Calendar Month',
+    value: { from: 'now/M', to: 'now' },
+  },
+  {
+    type: 'present',
+    label: 'This Calendar Quarter',
+    value: { from: 'thisQuarter', to: 'thisQuarter' },
+  },
+  {
+    type: 'present',
+    label: 'This Calendar Year',
+    value: { from: 'now/y', to: 'now' },
+  },
+  {
+    type: 'future',
+    label: 'Next Calendar Month',
+    value: { from: 'now+1M/M', to: 'now+2M/M' },
+  },
+  {
+    type: 'future',
+    label: 'Next Calendar Quarter',
+    value: { from: 'nextQuarter', to: 'nextQuarter' },
+  },
+  {
+    type: 'future',
+    label: 'Next Calendar Year',
+    value: { from: 'now+1y/y', to: 'now+2y/y' },
+  },
+  {
+    type: 'future',
+    label: 'Next 30 Days',
+    value: { from: 'now/d', to: 'now/d+30d' },
+  },
+  {
+    type: 'future',
+    label: 'Next 12 Months',
+    value: { from: 'now/d', to: 'now/d+12M' },
+  },
+  {
+    type: 'future',
+    label: 'Next 24 Months',
+    value: { from: 'now/d', to: 'now/d+24M' },
+  },
+  {
+    type: 'future',
+    label: 'Next 36 Months',
+    value: { from: 'now/d', to: 'now/d+36M' },
+  },
+]
+
+let rollingOptIsSelected = (node, opt) =>
+  node.from === opt.value.from && node.to === opt.value.to
+
+let toHTML5DateString = (node, endpoint) =>
```

Comment:
this seems like the signature should just take the date value directly - no point in overcomplicating it to actually lookup a prop on an object, just pass node.to/node.from inline